cmd/csi-snapshotter/main.go

```diff
@@ -155,10 +156,9 @@ func main() {
 	klog.V(2).Infof("CSI driver name: %q", *snapshotterName)
 
 	// Check it's ready
-	if err = csirpc.ProbeForever(csiConn, csiTimeout); err != nil {
+	if err = csirpc.ProbeForever(csiConn, *csiTimeout); err != nil {
```

Comment:
There is still just one timeout in the main. 

If you take a look of https://github.com/kubernetes-csi/external-attacher/blob/master/cmd/csi-attacher/main.go#L124 and https://github.com/kubernetes-csi/external-attacher/blob/master/cmd/csi-attacher/main.go#L131, there are two different timeouts.  In ProbeForever, it uses "timeout".  In context.WithTimeout, it uses csiTimeout.

Also the sequence is different.  In attacher, ProbeForever is called before context.WithTimeout and GetDriverName.  Should we do the same here?

Reply:
Maybe I'm missing something here. I think that in the other sidecars, those two values should be the same as well. For external-attacher, it seems like `csiTimeout` is only used for `GetDriverName` which seems odd to me. I can update the other sidecars to always use `--timeout` for ProbeForever &  GetDriverName. I think I also made the code harder to read because I assigned `csiTimeout` as the flag var where as in other places it's just assigned to `timeout`.

Reply:
Having one timeout for all operations seems reasonable to me

Reply:
sgtm! This PR as-is reflects that. I will follow-up in the other sidecars to do the same